src/collections/maple_tree/mod.rs

```diff
@@ -76,9 +77,243 @@ struct Node<K, V> {
     inner: NodeTag<K, V>,
 }
 
+#[derive(Debug, PartialEq)]
+pub enum FoundRange{
+        Min(usize), // this range goes from min to the index at K
```

Comment:
You could consider just using https://doc.rust-lang.org/std/ops/struct.Range.html and for min/max/etc just return the provided usize in those places.

IE FoundRange(min) would be Range(min, k). Max would be Range(k, max). MinMax would be Range(min, max). pivots would be Rnage(k1, k2).

Reply:
I think this could work but instead of using min and max's value we use -1 and R_CAPACITY because if min or max is equal to one of the values within the range 0..R_CAPACITY then we can't differentiate it from an index so we don't know if it's meant to be an index or if its meant to be min or max.

Reply:
Ahhhhh this is returning the INDEX of the pivots, not the range that is stored between the pivots. I see now.

Reply:
yeah sorry I need to work on my communication in the code.